packages/yoshi-common/src/user-actions-watcher.ts

```diff
@@ -0,0 +1,37 @@
+import readline from 'readline';
+
+type Callback = () => void;
+
+export class UserActionsWatcher {
+  private actions: { [key: string]: Callback } = {};
+
+  on(key: string, callback: Callback) {
+    this.actions[key] = callback;
+  }
+
+  watch() {
+    readline.emitKeypressEvents(process.stdin);
+
+    process.stdin.setRawMode(true);
+    process.stdin.setEncoding('utf8');
+
+    const CTRL_C = '\u0003';
+
+    process.stdin.on('keypress', (str) => {
+      if (str === CTRL_C) {
+        process.stdin.setRawMode(false);
+        process.exit(0);
```

Comment:
Obviously not critical, but this should be `SIGINT` code (128+2)

```suggestion
        process.exit(130);
```